### PR TITLE
Fix issue where the new calldata constraints are overwritten.

### PR DESCRIPTION
**Pull request by nbanmp**
Don't know how I missed this earlier. :disappointed: The calldatasize constraints are being applied every time now thanks to a `+=` instead of a `=`.

### CODE REVIEW

File: mythril/laser/ethereum/state.py [hunk 1]
```diff
@@ -74,11 +73,13 @@ def constraints(self):
             constraints.append(
                 ForAll(x, Implies(self[x] != 0, UGT(self.calldatasize, x)))
             )
-
+            #  10000 > calldatasize, prevents impossibly large calldatasizes
+            constraints.append(UGT(10000, self.calldatasize))
```
Comment (JoranHonig) [inline mythril/laser/ethereum/state.py]: I don't think this is the right place to add this, as its "technically" an untrue constraint. I'd rather put it in the concretization logic

Comment (JoranHonig) [reply mythril/laser/ethereum/state.py]: Also, this doesn't have to a part of this pr right?

Comment (nbanmp) [reply mythril/laser/ethereum/state.py]: You're right.

I was thinking that having a massive calldata could break other things, but I guess it only really breaks concretization.

File: tests/laser/transaction/symbolic_test.py [hunk 2]
```diff
@@ -67,3 +70,21 @@ def test_execute_contract_creation(mocked_setup: MagicMock):
...
     assert len(laser_evm.open_states) == 0
+
+
+def test_calldata_constraints_in_transaction():
+    # Arrange
+    laser_evm = LaserEVM({})
+    world_state = WorldState()
+
+    transaction = MessageCallTransaction(
+        world_state, Account("ca11ee"), Account("ca114")
```
Comment (JoranHonig) [inline tests/laser/transaction/symbolic_test.py]: Can you put in a Mocked calldata, where the function constraints returns a defined set of constraints. 
At assert we can then check if those specific constraints get added.
WDYT?

Comment (nbanmp) [reply tests/laser/transaction/symbolic_test.py]: Alright, I've updated the test case.

**Comment by b-mueller** [comment]
With this PR and #619 merged Mythril hangs on the [suicide multitx sample](https://github.com/SmartContractSecurity/SWC-registry/blob/master/test_cases/unprotected_critical_functions/suicide_multitx_feasible.sol) (or at least it gets stuck for 10+ minutes):

```
$MYTH -msuicide -v2 -x suicide_multitx_feasible.sol

root[82630] INFO Finished symbolic execution
root[82630] INFO 227 nodes, 226 edges, 1499 total states
root[82630] INFO Achieved 99.33774834437085 coverage for code: 6080604052600436106053576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff16806306661abd146058578063a444f5e9146080578063e1c7392a1460aa575b600080fd5b348015606357600080fd5b50606a60be565b6040518082815260200191505060405180910390f35b348015608b57600080fd5b5060a86004803603810190808035906020019092919050505060c4565b005b34801560b557600080fd5b5060bc60ed565b005b60015481565b60008054141560d15760ea565b3373ffffffffffffffffffffffffffffffffffffffff16ff5b50565b60016000819055505600a165627a7a7230582066d870cd55b625c157e8a6645ee3f810c25007c969f2dc4e303bbe7b22cfebb90029
root[82630] INFO Starting analysis
root[82630] INFO Executing <module 'suicide' from '/Users/bernhardmueller/Projects/mythril-classic/mythril/analysis/modules/suicide.py'>
root[82630] DEBUG Executing module: UNCHECKED_SUICIDE
root[82630] DEBUG [UNCHECKED_SUICIDE] suicide in function run(uint256)
root[82630] DEBUG 1_calldata
root[82630] DEBUG call_value1
root[82630] DEBUG call_value5
root[82630] DEBUG 2_calldatasize
root[82630] DEBUG 5_calldatasize
root[82630] DEBUG creator1
root[82630] DEBUG 1_calldatasize
root[82630] DEBUG call_value2
root[82630] DEBUG caller2
root[82630] DEBUG 2_calldata
root[82630] DEBUG 5_calldata
root[82630] DEBUG caller5
root[82630] DEBUG k!32
root[82630] DEBUG k!34
root[82630] DEBUG k!32!37
root[82630] DEBUG k!31
root[82630] DEBUG k!33
root[82630] DEBUG k!31!38
root[82630] DEBUG k!35
```

**Pull request merged**